alibi_detect/saving/tests/datasets.py

```diff
@@ -63,4 +65,7 @@ def data_synthetic_nd(data_shape):
 class TextData:
     @staticmethod
     def movie_sentiment_data():
-        return get_movie_sentiment_data()
+        try:
+            return get_movie_sentiment_data()
+        except RequestException:
+            pytest.skip('Movie sentiment dataset URL down')
```

Comment:
How does this actually work? The test within which this data is used is magically skipped?

Reply:
Embarrassingly I actually don't quite understand why this works. But I tested it by changing `MOVIESENTIMENT_URLS` to contain incorrect url's, and running `pytest alibi_detect/saving/tests/test_saving.py -k test_save_preprocess_nlp -sv`, and `SKIPPED (Movie sentiment dataset URL down)` is returned. Magic indeed! 

I think this is a special feature of the "cases" from `pytest-cases` that we use to represent datasets in the new saving tests. They're like fancy fixtures, and have clever capabilities like being able to use [pytest marks](https://smarie.github.io/python-pytest-cases/#pytest-marks-skip-xfail) (i.e. `skip`).